m2cgen/interpreters/python/interpreter.py

```diff
@@ -17,6 +19,23 @@ def interpret(self, expr):
             last_result = self._do_interpret(expr)
             self._cg.add_return_statement(last_result)
 
+        if self.with_linear_algebra or self.with_vectors:
+            self._cg.prepend_code_line("import numpy as np")
```

Comment:
Code leak into interpreter? I thought we should delegate all code related logic to Code Generator instead.